### PR TITLE
x/tokenfactory: cli

### PR DESCRIPTION
**Pull request by sunnya97**
Closes: #XXX

## What is the purpose of the change

*Add a description of the overall background and high level changes that this PR introduces *

*(E.g.: This pull request improves documation of area A by adding ....*

## Brief change log

*(for example:)*
 
  - *The metadata is stored in the blob store on job creation time as a persistent artifact*
  - *Deployments RPC transmits only the blob storage reference*
  - *Daemons retrieve the RPC data from the blob cache*

## Testing and Verifying

*(Please pick one of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added unit test that validates ...*
  - *Added integration tests for end-to-end deployment with ...*
  - *Extended integration test for ...*
  - *Manually verified the change by ...*

## Documentation and Release Note

  - Does this pull request introduce a new feature or user-facing behavior changes? (yes / no)
  - Is a relevant changelog entry added to the `Unreleased` section in `CHANGELOG.md`? (yes / no)
  - How is the feature or change documented? (not applicable   /   specification (`x/<module>/spec/`)  /  [Osmosis docs repo](https://github.com/osmosis-labs/docs)   /   not documented)

### CODE REVIEW

**Comment by faddat** [comment]
instead of cw20, or in addition to?

**Comment by sunnya97** [comment]
Tokenfactory is intended to be a replacement for cw20.  Instead of using cw20 tokens, CosmWasm contracts on Osmosis can mint/burn native sdk coins

File: x/tokenfactory/client/cli/query.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+	"github.com/osmosis-labs/osmosis/v7/x/tokenfactory/types"
+)
+
+// GetQueryCmd returns the cli query commands for this module
+func GetQueryCmd(queryRoute string) *cobra.Command {
+	// Group tokenfactory queries under a subcommand
+	cmd := &cobra.Command{
+		Use:                        types.ModuleName,
+		Short:                      fmt.Sprintf("Querying commands for the %s module", types.ModuleName),
```
Comment (p0mvn) [inline x/tokenfactory/client/cli/query.go:24]:
```suggestion
		Short:                      fmt.Sprintf("Querying commands for the %s module", queryRoute),
```

File: x/tokenfactory/client/cli/query.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+			return clientCtx.PrintProto(res)
+		},
+	}
+
+	flags.AddQueryFlagsToCmd(cmd)
+
+	return cmd
+}
+
+// GetCmdDenomAuthorityMetadata returns the authority metadata for a queried denom
```
Comment (p0mvn) [inline x/tokenfactory/client/cli/query.go:67]: The comment seems to be copied over from another function, please update

File: x/tokenfactory/module.go [hunk 3]
```diff
@@ -84,14 +85,12 @@ func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *r
 
 // GetTxCmd returns the capability module's root tx command.
```
Comment (p0mvn) [inline x/tokenfactory/module.go:86]:
```suggestion
// GetTxCmd returns the tokenfactory module's root tx command.
```

File: x/tokenfactory/module.go [hunk 4]
```diff
@@ -84,14 +85,12 @@ func (AppModuleBasic) RegisterGRPCGatewayRoutes(clientCtx client.Context, mux *r
...
 func (a AppModuleBasic) GetTxCmd() *cobra.Command {
-	// return cli.GetTxCmd()
-	return nil
+	return cli.GetTxCmd()
 }
 
 // GetQueryCmd returns the capability module's root query command.
```
Comment (p0mvn) [inline x/tokenfactory/module.go:91]:
```suggestion
// GetQueryCmd returns the tokenfactory module's root query command.
```

File: x/tokenfactory/client/cli/tx.go [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+// 			return tx.GenerateOrBroadcastTxWithFactory(clientCtx, txf, msg)
+// 		},
+// 	}
+
+// 	flags.AddTxFlagsToCmd(cmd)
+// 	return cmd
+// }
+
+// NewForceTransferCmd broadcast MsgForceTransfer
```
Comment (p0mvn) [inline x/tokenfactory/client/cli/tx.go:165]:
```suggestion
// NewChangeAdminCmd broadcast MsgChangeAdmin
```

File: x/tokenfactory/client/cli/tx.go [hunk 6]
```diff
@@ -0,0 +1,191 @@
...
+
+			return tx.GenerateOrBroadcastTxWithFactory(clientCtx, txf, msg)
+		},
+	}
+
+	flags.AddTxFlagsToCmd(cmd)
+	return cmd
+}
+
+// // NewForceTransferCmd broadcast MsgForceTransfer
```
Comment (p0mvn) [inline x/tokenfactory/client/cli/tx.go:131]: q: what is the plan for this command? When are we planning to enable it?

**Review by p0mvn — CHANGES_REQUESTED** [review]
Nice work! Left a few comments, please take a look

File: x/tokenfactory/client/cli/query.go (hunk 1 above)
Comment (ValarDragon) [reply x/tokenfactory/client/cli/query.go:24]: I think we should keep types.ModuleName, and delete queryRoute

File: x/tokenfactory/client/cli/tx.go (hunk 6 above)
Comment (sunnya97) [reply x/tokenfactory/client/cli/tx.go:129]: We have to conduct an audit to make sure that doing a forceTransfer of coins from an account, especially a module account won't cause panics

https://github.com/osmosis-labs/feature-requests/issues/73

**Comment by sunnya97** [comment]
Please merge #1435 first

**Review by ValarDragon — APPROVED** [review]
LGTM, I'm looking forward to generalized tooling to delete all of this.

All the CLI code is pure boilerplate given a typed go struct lol

**Comment by sunnya97** [comment]
Hopefully telescope fixes this

**Pull request merged**